src/pages/UserHome.jsx

```diff
@@ -25,6 +31,27 @@ export default function UserHome() {
       <Helmet>
         <title>Dashboard</title>
       </Helmet>
+
+      {location?.state?.name && (
```

Comment:
Hmm, this is the sort of thing I might envision most other pages to have (including the other PR). But from reading this code it's not straightforward what this refers to. What if we instead make a `state.alert` object of this format?
```js
{
  action: {
    name: "Try again",
    url: "some url",
  },
  message: "message to show",
  severity: "error",
}
```

Something like this could probably be generalized for other similar use cases too.

Reply:
If we don't redirect when there is an error, this can simple be `{message: "message to show"}`